content/blog/fast-data-processing-pipeline-for-predicting-flight-delays-using-apache-apis-kafka-spark-streaming-and-machine-learning-part-2.md

```diff
@@ -0,0 +1,231 @@
+
+![Create new view, index, cache](/img/create-new.png)
+
+![Read From new View](/img/read-from-new-view.png)
+
+## Data Pipelines
+
+When you combine these messaging capabilities with the simple concept of microservices, you can greatly enhance the agility with which you build, deploy, and maintain complex data pipelines. Pipelines are constructed by simply chaining together multiple microservices, each of which listens for the arrival of some data, performs its designated task, and optionally publishes its own messages to a topic. Development teams can deploy new services or service upgrades more frequently and with less risk, because the production version does not need to be taken offline. Both versions of the service simply run in parallel, consuming new data as it arrives and producing multiple versions of output. Both output streams can be monitored over time; the older version can be decommissioned when it ceases to be useful.
+
+## Machine Learning Logistics and Data Pipelines
+
+Combining data pipelines with machine learning can handle the logistics of machine learning in a flexible way by:
+
+* Making input and output data available to independent consumers
+* Managing and evaluating multiple models and easily deploying new models
+
+![Machine Learning Logistics and Data Pipelines](/img/machine-learning-logistics.png)
+
+Architectures for these types of applications are discussed in more detail in the ebooks Machine Learning logistics, Streaming Architecture, and Microservices and Containers.
+
+Below is the data processing pipeline for this use case of predicting flight delays. This pipeline could be augmented to be part of the rendezvous architecture discussed in the Oreilly [Machine Learning Logistics ebook](https://www.oreilly.com/library/view/machine-learning-logistics/9781491997628/).
+
+![Data Processing Pipeline](/img/data-processing-pipeline.png)
+
+## Spark Streaming Use Case Example Code
+
+The following figure depicts the architecture for the part of the use case data pipeline discussed in this post:
+
+![architecture data pipeline](/img/architechture-data-pipeline.png)
+
+1. ```
+      Flight data is published to a MapR Event Store topic using the Kafka API.
+   ```
+2. ```
+      A Spark streaming application, subscribed to the first topic:
+   ```
+3. ```
+      Ingests a stream of flight data
+   ```
+4. ```
+      Uses a deployed machine learning model to enrich the flight data with a delayed/not delayed prediction
+   ```
+5. ```
+      publishes the results in JSON format to another topic.
+   ```
+6. ```
+      (In the 3rd blog) A Spark streaming application subscribed to the second topic:
+   ```
+7. ```
+      Stores the input data and predictions in MapR Database
+   ```
+
+## Example Use Case Data
+
+You can read more about the data set in part 1 of this series. The incoming and outgoing data is in JSON format, an example is shown below:
```

Comment:
Recommend hot-linking to part 1 here.

![image](https://user-images.githubusercontent.com/77496261/137188121-525a6aaf-3099-4fc5-85a0-91d503bbc710.png)

Reply:
done